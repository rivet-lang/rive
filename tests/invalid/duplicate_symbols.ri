fn main() {
    let x = 1;
    let x = 1; // FAIL

    let y = 2;
    {
        let y = 5; // disallow variable shadowing
    }
}

fn other(a: i32) {
    let a = 1;
}

fn other(a: i32) { // FAIL
    let a = 1;
}

class RuntimeError : Error { }
class RuntimeError : Error { } // FAIL

struct Gota { }
struct Gota { } // FAIL

enum State {
    Idle,
    Idle // FAIL
}
