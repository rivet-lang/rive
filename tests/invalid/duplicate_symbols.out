tests/invalid/duplicate_symbols.ri:15:4: error: module `duplicate_symbols` has duplicate symbol `other`
  15 | fn other(a: i32) { // FAIL
     |    ^
tests/invalid/duplicate_symbols.ri:20:7: error: module `duplicate_symbols` has duplicate symbol `RuntimeError`
  20 | class RuntimeError : Error { } // FAIL
     |       ^
tests/invalid/duplicate_symbols.ri:23:8: error: module `duplicate_symbols` has duplicate symbol `Gota`
  23 | struct Gota { } // FAIL
     |        ^
tests/invalid/duplicate_symbols.ri:25:6: error: enum `State` has duplicate value `Idle`
  25 | enum State {
     |      ^
rivetc: error: could not compile module `duplicate_symbols`, aborting due to 4 previous errors